pkg/cli/mirror/catalog_images.go

```diff
@@ -85,6 +109,14 @@ func (o *MirrorOptions) rebuildCatalogs(ctx context.Context, dstDir string) (ima
 			ctlgRef := image.TypedImage{}
 			ctlgRef.Type = imagesource.DestinationRegistry
 			sourceRef, err := image.ParseReference(img)
+			// since we can't really tell if the "img" reference originated from an actual docker
+			// reference or from an OCI file path that approximates a docker reference, ParseReference
+			// might not lowercase the name and namespace values which is required by the
+			// docker reference spec (see https://github.com/distribution/distribution/blob/main/reference/reference.go).
+			// Therefore we lower case name and namespace here to make sure it's done.
+			sourceRef.Ref.Name = strings.ToLower(sourceRef.Ref.Name)
```

Comment:
If things are done right, sourceRef.OCIFBCPath should not be empty  and this can be a good indication that the image is OCI local

Reply:
### Clarification

Actually in this case, the sourceRef comes from the `img` variable, which is the "repoPath" which could be:

- A valid docker ref reconstituted from the file system path
  - example: `foo.com/foo/bar/<id>`
- Some path for a OCI layout
  - example: `some/path/provided/by/user/<id>`

Because `img` does not have `oci:` prefix, the `ParseReference` function is going to parse this "repoPath" value as a docker reference, and the `sourceRef.OCIFBCPath` is never going to be populated.

### Discussion for how `sourceRef.OCIFBCPath` could be properly populated
So even if you wanted to add the `oci:` prefix to `img`, how would you know when to do so? Looking at the examples above, there's only one way to distinguish the difference between the two and you'd have to process the assumed "domain" portion of the path (please refer to the BNF below for terminology). You'd have to start making assumptions that if the first component in the path contains `.` or `:` then it's a docker ref and otherwise it's an OCI path.  

```
domain                          := host [':' port-number]
host                            := domain-name | IPv4address | \[ IPv6address \]	; rfc3986 appendix-A
domain-name                     := domain-component ['.' domain-component]*
domain-component                := /([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9])/
```

This gets tricky as the examples show:

Examples that are easy enough to distinguish as docker refs based on just the "domain":
`foo.com/foo/bar/<id>`
`1.2.3.4/foo/bar/<id>`
`::ffff:102:304/foo/bar/<id>`

Examples that have easy workarounds you can hard code to test for (e.g. localhost):
`localhost/foo/bar/<id>`

Examples that the user could do and you'd have no way to tell they are valid domains or not:
`my.example/foo/bar/<id>`
`my:example/foo/bar/<id>`

I'd worry about false positives for detecting a valid docker reference given the last two examples.